use-cases/LOC-test-program/test-main.cpp

```diff
@@ -0,0 +1,40 @@
+/**
+ * *****************************************************************************
+ * \file test-main.cpp
+ * \author Aditya P. Gurajajada
+ * \brief L3_LOC integration unit-test program.
+ * \version 0.1
+ * \date 2024-04-14
+ *
+ * All this program does is to assert that the expected typeid() for __LOC__
+ * matches the expected type-ID for defined base typedef. This 'test' program
+ * exists mainly to debug-through some build issues encountered while
+ * developing the integration of the machinery for LOC-ELF encoding with L3's
+ * logging APIs.
+ *
+ * \copyright Copyright (c) 2024
+ * *****************************************************************************
+ */
+#include <iostream>
+#include <assert.h>
+
+#include "l3.h"
+
+using namespace std;
+
+int
+main(const int argc, const char *argv[])
+{
+    assert(typeid(__LOC__) == typeid(intptr_t));
```

Comment:
NOTE: This is a testing assertion, just to figure out what's going on.

Currently, in LOC package `__LOC__` is defined with LOC-ELF encoding to return an expression of type `intptr_t`. We are effectively casting it down to int32_t to save-it off as the index of the LOC-ELF section's  entries.

This works for now, but is a bit error prone. 

In a future rev, I plan to change LOC package to return this casted to `int32_t`, at which time this assert will flip.

And so will the asserts on L33 and L36 below. They will both change to `==`.

That's a change coming in the future.